src/SourceBuild/content/repo-projects/package-source-build.proj

```diff
@@ -7,63 +7,53 @@
       <SourceBuiltVersionFileName>.version</SourceBuiltVersionFileName>
     </PropertyGroup>
 
-    <!-- Content of the .version file to include in the tarball. Write to staging dir. -->
+    <!-- Content of the .version file to include in the tarball -->
     <ItemGroup>
       <VersionFileContent Include="$(RepositoryCommit);$(SourceBuiltSdkVersion)" />
     </ItemGroup>
 
     <WriteLinesToFile
-      File="$(SourceBuildTarballStagingDir)$(SourceBuiltVersionFileName)"
+      File="$(BaseIntermediateOutputPath)$(SourceBuiltVersionFileName)"
       Lines="@(VersionFileContent)"
       Overwrite="true" />
 
-    <MakeDir Directories="$([System.IO.Path]::GetDirectoryName('$(SourceBuiltTarballName)'))" />
-    <Exec Command="tar --numeric-owner -czf $(SourceBuiltTarballName) $(SourceBuiltVersionFileName) *.nupkg *.props SourceBuildReferencePackages/"
-          WorkingDirectory="$(SourceBuildTarballStagingDir)" />
+    <MakeDir Directories="$(SharedOutputPath)" />
 
-    <Message Importance="High" Text="Packaged source-built artifacts to $(SourceBuiltTarballName)" />
+    <ItemGroup>
+      <_AllRepoDirs Include="$([System.IO.Directory]::GetDirectories($(ArtifactsShippingPackagesDir)))" />
+      <_AllRepoDirs Include="$([System.IO.Directory]::GetDirectories($(ArtifactsNonShippingPackagesDir)))" />
+    </ItemGroup>
+
+    <!-- Create a layout directory for the files that are to be included in the artifacts tarball. Since there are a large number of files, this will use symlinks
+         instead of copying files to make this execute quickly. -->
+    <PropertyGroup>
+      <_SourceBuiltLayoutDir>$(BaseIntermediateOutputPath)artifacts-layout/</_SourceBuiltLayoutDir>
+      <SourceBuildReferencePackagesDestinationDirName>SourceBuildReferencePackages</SourceBuildReferencePackagesDestinationDirName>
+    </PropertyGroup>
+    <MakeDir Directories="$(_SourceBuiltLayoutDir)$(SourceBuildReferencePackagesDestinationDirName)" />
+    <Exec Command="ln -sf %(_AllRepoDirs.Identity)/* $(_SourceBuiltLayoutDir)"
+          Condition="$([System.IO.Directory]::GetFiles('%(_AllRepoDirs.Identity)')) != ''" />
+    <Exec Command="ln -sf $(BaseIntermediateOutputPath)$(SourceBuiltVersionFileName) $(_SourceBuiltLayoutDir)" />
+    <Exec Command="ln -sf $(CurrentSourceBuiltPackageVersionPropsPath) $(_SourceBuiltLayoutDir)PackageVersions.props" />
+    <Exec Command="ln -sf $(ReferencePackagesDir)* $(_SourceBuiltLayoutDir)$(SourceBuildReferencePackagesDestinationDirName)" />
+
+    <Exec Command="tar --numeric-owner -czhf $(SourceBuiltTarballName) $(SourceBuiltVersionFileName) *"
+          WorkingDirectory="$(_SourceBuiltLayoutDir)" />
 
-    <RemoveDir Directories="$(SourceBuildTarballStagingDir)" />
+    <Message Importance="High" Text="Packaged source-built artifacts to $(SourceBuiltTarballName)" />
   </Target>
 
+  <ItemGroup>
+    <RepositoryReference Include="installer" />
+    <RepositoryReference Include="scenario-tests" />
+  </ItemGroup>
```

Comment:
Please move these up, before the target where they were before.

Reply:
Fixed in https://github.com/dotnet/installer/pull/18557/commits/a7a6f797cf0c959a333b8362958a0873b3540f5b